From: Mike Travis <travis@sgi.com>
Subject: x86: Reduce per cpu warning boot up messages
References: bnc#587669, FATE#306952
Patch-mainline: not yet

Reduce warning message output to one line only instead of per
cpu.

Signed-of-by: Mike Travis <travis@sgi.com>
Signed-off-by: Ingo Molnar <mingo@elte.hu>
Signed-off-by: Rafael J. Wysocki <rjw@suse.de>
Automatically created from "patches.arch/x86-Reduce-per-cpu-warning-boot-up-messages.patch" by xen-port-patches.py

--- sle11sp1-2010-04-12.orig/arch/x86/kernel/process-xen.c	2010-02-09 17:13:11.000000000 +0100
+++ sle11sp1-2010-04-12/arch/x86/kernel/process-xen.c	2010-04-12 14:53:21.000000000 +0200
@@ -524,7 +524,7 @@ void __cpuinit select_idle_routine(const
 #ifndef CONFIG_XEN
 #ifdef CONFIG_SMP
 	if (pm_idle == poll_idle && smp_num_siblings > 1) {
-		printk(KERN_WARNING "WARNING: polling idle and HT enabled,"
+		printk_once(KERN_WARNING "WARNING: polling idle and HT enabled,"
 			" performance may degrade.\n");
 	}
 #endif
